src/rules/at-rule-no-unknown/index.js

```diff
@@ -0,0 +1,49 @@
+import { isString } from "lodash"
+import { vendor } from "postcss"
+import {
+  optionsHaveIgnoredAtRule,
+  report,
+  ruleMessages,
+  validateOptions,
+} from "../../utils"
+import { atRules } from "../../reference/keywordSets"
+
+export const ruleName = "at-rule-no-unknown"
+
+export const messages = ruleMessages(ruleName, {
+  rejected: (atRule) => `Unexpected unknown at-rule "${atRule}"`,
+})
+
+export default function (actual, options) {
+  return (root, result) => {
+    const validOptions = validateOptions(result, ruleName, { actual }, {
+      actual: options,
+      possible: {
+        ignoreAtRules: [isString],
+      },
+      optional: true,
+    })
+
+    if (!validOptions) { return }
+
+    root.walkAtRules(atRule => {
+      const nameAtRule = atRule.name
+
+      if (vendor.prefix(nameAtRule) || atRules.has(nameAtRule.toLowerCase())) { return }
+
+      if (optionsHaveIgnoredAtRule(options, atRule)) { return }
```

Comment:
Lets remove this line and just support an array of strings, rather than regex too, to keep it consistent with the other `*-no-unknown` rules. If we think the `*-no-unknown` rules would benefit from regex support, then we can add that in another PR.